$OpenBSD: patch-uitoolkit_fb_ui_font_c,v 1.1 2017/08/23 16:10:37 dcoppa Exp $

Fix wrong processing of percentage in a font name on framebuffer
(upstream HG changeset 3b790a86946a9338f481f894018e9855e5ae94c7)

Index: uitoolkit/fb/ui_font.c
--- uitoolkit/fb/ui_font.c.orig
+++ uitoolkit/fb/ui_font.c
@@ -31,6 +31,7 @@
 
 #define DIVIDE_ROUNDING(a, b) (((int)((a)*10 + (b)*5)) / ((int)((b)*10)))
 #define DIVIDE_ROUNDINGUP(a, b) (((int)((a)*10 + (b)*10 - 1)) / ((int)((b)*10)))
+#define DIVIDE_ROUNDINGDOWN(a, b) (((int)(a)*10) / ((int)((b)*10)))
 
 #ifdef WORDS_BIGENDIAN
 #define _TOINT32(p, is_be) ((is_be) ? TOINT32(p) : LE32DEC(p))
@@ -688,7 +689,7 @@ face_found:
   xfont->face = face;
   xfont->is_aa = is_aa;
 
-  if (!load_glyph(face, format, get_glyph_index(face, 'W'), is_aa)) {
+  if (!load_glyph(face, format, get_glyph_index(face, 'M'), is_aa)) {
     bl_msg_printf("%s doesn't have outline glyphs.\n", file_path);
 
     goto error;
@@ -1704,14 +1705,10 @@ ui_font_t *ui_font_new(Display *display, vt_font_t id,
 #ifdef USE_FREETYPE
 #ifdef USE_FONTCONFIG
   if (use_fontconfig && dpi_for_fc > 0.0) {
-    fontsize = DIVIDE_ROUNDINGUP(dpi_for_fc * fontsize, 72);
+    fontsize = DIVIDE_ROUNDINGDOWN(dpi_for_fc * fontsize, 72);
   }
 #endif
-  if (percent > 0) {
-    format = DIVIDE_ROUNDING(fontsize * percent, 100) | (id & (FONT_BOLD | FONT_ITALIC));
-  } else {
-    format = fontsize | (id & (FONT_BOLD | FONT_ITALIC));
-  }
+  format = fontsize | (id & (FONT_BOLD | FONT_ITALIC));
 #endif
 
   if (decsp_id) {
